src/commands/serve.rs

```diff
@@ -373,10 +373,14 @@ impl hyper::service::Service<Request> for ProxyHandler {
                 wasmtime_wasi_http::proxy::Proxy::instantiate_pre(&mut store, &inner.instance_pre)
                     .await?;
 
-            proxy
+            if let Err(e) = proxy
                 .wasi_http_incoming_handler()
                 .call_handle(store, req, out)
-                .await?;
+                .await
+            {
+                eprint!("{:?}\n", e);
+                return Err(e);
+            }
 
             Ok::<_, anyhow::Error>(())
```

Comment:
You can probably remove the annotation on `Ok`, now that there's an explicit `Err` use.

Reply:
```suggestion
            Ok(())
```